sqlmesh/core/model/meta.py

```diff
@@ -462,6 +520,35 @@ def custom_materialization_properties(self) -> CustomMaterializationProperties:
             return self.kind.materialization_properties
         return {}
 
+    @cached_property
+    def grants(self) -> t.Optional[GrantsConfig]:
+        """A dictionary of grants mapping permission names to lists of grantees."""
+
+        if not self.grants_:
+            return None
+
+        def parse_exp_to_str(e: exp.Expression) -> str:
+            if isinstance(e, exp.Literal) and e.is_string:
+                return e.this.strip()
+            if isinstance(e, exp.Identifier):
+                return e.name
+            return e.sql(dialect=self.dialect).strip()
+
+        grants_dict = {}
+        for eq_expr in self.grants_.expressions:
+            permission_name = parse_exp_to_str(eq_expr.this)  # left hand side
+            grantees_expr = eq_expr.expression  # right hand side
+            if isinstance(grantees_expr, exp.Array):
```

Comment:
If it's not array we just silently ignore it? Do we validate anywhere that the right-hand side is an array?